scripts/binary/meta.ts

```diff
@@ -0,0 +1,100 @@
+import path from 'path'
+import os from 'os'
+
+// canonical platform names
+export const platforms = {
+  darwin: 'darwin',
+  linux: 'linux',
+  windows: 'win32',
+} as const
+
+export const PLATFORM = os.platform() as any
+
+if (!Object.values(platforms).includes(PLATFORM)) {
+  throw new Error(`Invalid build platform  ${PLATFORM}`)
+}
+
+export type PlatformName = {[K in keyof typeof platforms]: typeof platforms[K]}[keyof typeof platforms]
+
+export const buildRootDir = () => {
+  return path.join(TMP_BUILD_DIR, 'build')
+}
+
+export const buildLinkDir = () => {
+  return path.resolve('build')
+}
+
+// returns a path into the /build directory
+// the output folder should look something like this
+// build/
+//   <platform>/ = linux or darwin
+//     ... platform-specific files
+export const buildDir = function (...args: string[]) {
+  const root = buildRootDir()
+
+  switch (PLATFORM) {
+    case 'darwin':
+      // the new electron-builder for some reason adds its own platform
+      // subfolder and it is NOT "darwin" but "mac"
+      return path.resolve(root, 'mac', ...args)
+    case 'linux':
+      return path.resolve(root, 'linux-unpacked', ...args)
+    case 'win32':
+      if (os.arch() === 'x64') {
+        return path.resolve(root, 'win-unpacked', ...args)
+      }
+
+      // x86 32bit architecture
+      return path.resolve(root, 'win-ia32-unpacked', ...args)
```

Comment:
windows 32 bit is not longer suppported: https://github.com/cypress-io/cypress/pull/17285/files#diff-5cb3255ebb2fdd25352410f9cbbc79d2096d86b27d3b4db8f7462f1ddd3ae45dL46-L47

Reply:
Good catch